### PR TITLE
Remove unused geoserver dockerfiles

### LINKED ISSUE #349: :shield: [Security] Remove deprecated GeoServer
## Description

Some Docker references for GeoServer are specified here: 
https://github.com/bird-house/birdhouse-deploy/tree/master/birdhouse/docker/geoserver

However, they have not been modified over 4 years, and only seem to be leftover and unused code. 
This should be removed entirely, as it causes unnecessary confusion about which GeoServer is employed, as well as causing many (>50 critical, >160 high) vulnerabilities to be reflected by security scans due to very old references.

The active GeoServer in the stack is referenced with the following file instead: 
https://github.com/bird-house/birdhouse-deploy/blob/master/birdhouse/config/geoserver/Dockerfile

A much more up-to-date and reduced set of security vulnerabilities (~4 critical with `pavics/geoserver:2.22.2`) is detailed in https://github.com/bird-house/birdhouse-deploy/issues/320 (note that this is still not the latest active version in the stack still using `pavics/geoserver:2.19.0).

## References

- relates to https://github.com/bird-house/birdhouse-deploy/issues/320

## Concerned Organizations

All using GeoServer.

### PR DESCRIPTION
**Pull request by tlvu**
- Delete unused Dockerfiles, fixes #349 and fixes #352

  * birdhouse/docker/geoserver: not used since 3-4 years, replaced by https://github.com/kartoza/docker-geoserver

  * birdhouse/config/geoserver/Dockerfile: was introduced in commit f3b9896e6b771e0aff62c6851c2376d730ddadaf (PR #233, merge commit d1ecc63284ec9d2940bfa2b1b4baca3fbe1308b3) as a temporary solution only, not needed with newer kartoza docker images.

- Move birdhouse/docker/solr to birdhouse/deprecated-components/solr/docker to group related files together.  Solr has been deprecated since PR #311 (merge commit a8d3612fdb7fd7758b24e75b0ef697fd3d8ace51).

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -15,7 +15,20 @@
 [Unreleased](https://github.com/bird-house/birdhouse-deploy/tree/master) (latest)
 ------------------------------------------------------------------------------------------------------------------
 
-[//]: # (list changes here, using '-' for each new entry, remove this when items are added)
+## Changes
+
+- Delete unused Dockerfiles, fixes #349 and #352
```
Comment (fmigneault) [inline CHANGES.md:20]: PR numbers and commit hash should use the full URL for generated docs (here and below)

File: CHANGES.md [hunk 2]
```diff
@@ -15,7 +15,20 @@
...
+
+  * birdhouse/docker/geoserver: not used since 3-4 years, replaced by https://github.com/kartoza/docker-geoserver
+
+  * birdhouse/config/geoserver/Dockerfile: was introduced in commit f3b9896e6b771e0aff62c6851c2376d730ddadaf
```
Comment (fmigneault) [inline CHANGES.md:24]: use code format for paths

File: CHANGES.md [hunk 3]
```diff
@@ -15,7 +15,20 @@
...
+    (PR #233, commit d1ecc63284ec9d2940bfa2b1b4baca3fbe1308b3) as a temporary
+    solution only, not needed with newer kartoza docker images.
+
+- Move birdhouse/docker/solr to birdhouse/deprecated-components/solr/docker
```
Comment (fmigneault) [inline CHANGES.md:28]: Use the path reference format (see example in below version).

**Review by fmigneault — COMMENTED** [review]
Just got some formatting issues in the change logs.

The features themselves are good.

**Review by mishaschwartz — APPROVED** [review]

**Comment by tlvu** [comment]
Not tagging since this do not have any active code change.  This is akin to simple documentation change.

**Comment by mishaschwartz** [comment]
@tlvu no version bump for this one?

**Pull request merged**